File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {

        int winsum = 0;
        for (int i = 0; i < k; i++) {
            winsum = winsum + nums[i];
        }

        int maxsum = winsum;
        for (int i = k; i < nums.size(); i++) {
            winsum = winsum + nums[i] - nums[i - k];
            maxsum = max(maxsum, winsum);
        }

        return (double)maxsum / k;
    }
};

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        double ans=Integer.MIN_VALUE;
        int size=nums.length;

        double sum=0;
        for(int i=0;i<k;i++){
            sum=sum+nums[i];
        }

        double avg=sum/k;
        ans=Math.max(ans,avg);
        int l=0;

        for(int i=k;i<size;i++){
            sum=sum+nums[i]-nums[l];
            l++;
            double new_avg=sum/k;
         
            ans=Math.max(ans,new_avg);
        }

        return ans;
    }
}